webapp/channels/src/components/suggestion/emoticon_provider.tsx

```diff
@@ -12,55 +12,55 @@ import {getEmojiMap, getRecentEmojisNames} from 'selectors/emojis';
 
 import store from 'stores/redux_store.jsx';
 
+import {Preferences} from 'utils/constants';
 import * as Emoticons from 'utils/emoticons';
 import {compareEmojis, emojiMatchesSkin} from 'utils/emoji_utils';
 
-import Suggestion from './suggestion.jsx';
-import Provider from './provider';
+import {SuggestionContainer, SuggestionProps} from './suggestion';
+import Provider, {ResultsCallback} from './provider';
 
 export const MIN_EMOTICON_LENGTH = 2;
 export const EMOJI_CATEGORY_SUGGESTION_BLOCKLIST = ['skintone'];
 
-class EmoticonSuggestion extends Suggestion {
-    render() {
-        const text = this.props.term;
-        const emoji = this.props.item.emoji;
-
-        let className = 'emoticon-suggestion';
-        if (this.props.isSelection) {
-            className += ' suggestion--selected';
-        }
+type EmojiItem = {
+    name: string;
+    emoji: Emoji;
+    type: string;
+}
 
-        return (
-            <div
-                className={className}
-                onClick={this.handleClick}
-                onMouseMove={this.handleMouseMove}
-                {...Suggestion.baseProps}
-            >
-                <div className='pull-left'>
-                    <img
-                        alt={text}
-                        className='emoticon-suggestion__image'
-                        src={getEmojiImageUrl(emoji)}
-                        title={text}
-                    />
-                </div>
-                <div className='pull-left'>
-                    {text}
-                </div>
+const EmoticonSuggestion = React.forwardRef<HTMLDivElement, SuggestionProps<EmojiItem>>((props, ref) => {
+    const text = props.term;
+    const emoji = props.item.emoji;
```

Comment:
0/5 but this could be rewritten as `const {text, item: {emoji}} = props;`

Not sure if more or less legible :sweat_smile:

Reply:
That one I think I'll leave as-is. I have a hard time reading nested destructuring and figuring out which variables are declared by it